frontend/src/components/note/NoteCard.vue

```diff
@@ -0,0 +1,101 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+
+import { Card, Divider } from '@/lib/primevue';
+import { userService } from '@/services';
+import { formatDate } from '@/utils/formatters';
+
+import type { Ref } from 'vue';
+import type { Note } from '@/types';
+
+// Props
+type Props = {
+  note: Note;
+  submissionId: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {});
+
+// State
+const cardData: Ref<Note> = ref(props.note);
```

Comment:
Unnecessary state value. You are never modifying the state. Just read from the prop instead.